be/src/storage/storage_engine.cpp

```diff
@@ -543,6 +544,43 @@ void StorageEngine::_start_clean_fd_cache() {
     VLOG(10) << "Cleaned file descriptor cache";
 }
 
+void StorageEngine::compaction_check() {
+    int checker_one_round_sleep_time_s = 1800;
+    while (true) {
+        MonotonicStopWatch stop_watch;
+        stop_watch.start();
+        LOG(INFO) << "start to check compaction";
+        size_t num = compaction_check_one_round();
+        stop_watch.stop();
+        LOG(INFO) << num << " tablets checked. time elapse:" << stop_watch.elapsed_time() / 1e9 << " seconds."
+                  << " compaction checker will be scheduled again in " << checker_one_round_sleep_time_s << " seconds";
+        std::this_thread::sleep_for(std::chrono::seconds(checker_one_round_sleep_time_s));
```

Comment:
This will make the exit gracefully stall on 1800 seconds.

Reply:
fixed